src/Analyser/MutatingScope.php

```diff
@@ -1129,9 +1129,6 @@ private function resolveType(string $exprString, Expr $node): Type
 				}
 
 				$resultType = $this->initializerExprTypeResolver->resolveConcatType($resultType, $partType);
-				if (count($resultType->getConstantStrings()) === 0) {
```

Comment:
We will still need to check every part to see if it's a lowercase string.

I'm unsure I should change this to
```
if (count($resultType->getConstantStrings()) === 0 && $resultType->isLowercaseString()) {
```
to avoid a similar bug to come back if we introduce another string Accessory.